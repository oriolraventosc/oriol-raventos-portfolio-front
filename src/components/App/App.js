import About from "../About/About";
import Header from "../Header/Header";
import Hero from "../Hero/Hero";
import Technologies from "../Technologies/Technologies";

const App = () => {
  return (
    <div className="App">
      <Header />
      <Hero />
      <About />
      <Technologies />
    </div>
  );
};

export default App;
