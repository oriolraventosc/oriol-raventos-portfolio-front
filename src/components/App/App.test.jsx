import TestRenderer from "react-test-renderer";
import App from "./App";
import { GlobalStyles } from "@mui/styled-engine";
import { ThemeProvider } from "styled-components";
import MainTheme from "../../styles/ThemeStyles";

describe("Given a App component", () => {
  describe("When it is rendered", () => {
    test("Then it should show the homepage", () => {
      const expected = TestRenderer.create(
        <ThemeProvider theme={MainTheme}>
          <GlobalStyles />
          <App />
        </ThemeProvider>
      );

      expect(expected).toMatchSnapshot();
    });
  });
});
