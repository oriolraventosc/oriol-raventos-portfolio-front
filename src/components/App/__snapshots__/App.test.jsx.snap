// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a App component When it is rendered Then it should show the homepage 1`] = `
<div
  className="App"
>
  <header
    className="sc-hLBbgP igvvhP"
  >
    <section>
      <div
        aria-label="main-menu"
        className="header-mobile"
      >
        <a
          aria-label="oriol-raventos-portfolio-top"
          href="#top-page"
          onClick={[Function]}
        >
          <span
            className="header-mobile__title"
          >
            { or }
          </span>
        </a>
        <div
          aria-label="hamburger-menu"
          className="hamburger"
        >
          <div
            aria-expanded={false}
            className="hamburger-react"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            style={
              Object {
                "cursor": "pointer",
                "height": "48px",
                "outline": "none",
                "position": "relative",
                "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                "userSelect": "none",
                "width": "48px",
              }
            }
            tabIndex={0}
          >
            <div
              style={
                Object {
                  "background": "#fff",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "17px",
                  "transform": "none",
                  "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
            <div
              style={
                Object {
                  "background": "#fff",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "23px",
                  "transform": "none",
                  "transition": "0.05s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
            <div
              style={
                Object {
                  "background": "#fff",
                  "height": "2px",
                  "left": "14px",
                  "position": "absolute",
                  "top": "29px",
                  "transform": "none",
                  "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                  "width": "20px",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        aria-label="portfolio-menu"
        className="header-mobile-navigation"
      >
        <div
          className="portfolio-menu__top"
        >
          <a
            aria-label="oriol-raventos-portfolio-top"
            href="#top-page"
            onClick={[Function]}
          >
            <span
              className="header-mobile__title"
            >
              { or }
            </span>
          </a>
          <div
            aria-label="hamburger-menu-close"
            className="hamburger"
          >
            <div
              aria-expanded={false}
              aria-label="toggle-menu-close"
              className="hamburger-react"
              onClick={[Function]}
              onKeyUp={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "height": "48px",
                  "outline": "none",
                  "position": "relative",
                  "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                  "userSelect": "none",
                  "width": "48px",
                }
              }
              tabIndex={0}
            >
              <div
                style={
                  Object {
                    "background": "#fff",
                    "height": "2px",
                    "left": "14px",
                    "position": "absolute",
                    "top": "17px",
                    "transform": "none",
                    "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                    "width": "20px",
                  }
                }
              />
              <div
                style={
                  Object {
                    "background": "#fff",
                    "height": "2px",
                    "left": "14px",
                    "position": "absolute",
                    "top": "23px",
                    "transform": "none",
                    "transition": "0.05s cubic-bezier(0, 0, 0, 1)",
                    "width": "20px",
                  }
                }
              />
              <div
                style={
                  Object {
                    "background": "#fff",
                    "height": "2px",
                    "left": "14px",
                    "position": "absolute",
                    "top": "29px",
                    "transform": "none",
                    "transition": "0.1s cubic-bezier(0, 0, 0, 1)",
                    "width": "20px",
                  }
                }
              />
            </div>
          </div>
        </div>
        <a
          className="header-mobile__navigation__item"
          href="#about me"
          onClick={[Function]}
        >
          about me
        </a>
        <a
          className="header-mobile__navigation__item"
          href="#projects"
          onClick={[Function]}
        >
          projects
        </a>
        <a
          className="header-mobile__navigation__item"
          href="#contact"
          onClick={[Function]}
        >
          contact
        </a>
      </div>
    </section>
    <section
      className="header-desktop"
    >
      <a
        aria-label="oriol-raventos-portfolio-top"
        href="#top-page"
      >
        <span
          className="header-desktop__title"
        >
          { oriol raventós }
        </span>
      </a>
      <div
        aria-label="portfolio-desktop-menu"
        className="header-desktop__navigation"
      >
        <a
          className="header-desktop__navigation__item"
          href="#about me"
        >
          about me
        </a>
        <a
          className="header-desktop__navigation__item"
          href="#projects"
        >
          projects
        </a>
        <a
          className="header-desktop__navigation__item"
          href="#contact"
        >
          contact
        </a>
      </div>
    </section>
  </header>
  <section
    aria-label="hero"
    className="sc-eDvSVe ebKfYP"
    id="top-page"
  >
    <div
      className="hero"
    >
      <p>
        Hello.
      </p>
      <h1>
        I am 
        <strong>
          oriol raventós.
        </strong>
      </h1>
      <div
        className="hero__typed-out"
      >
        <p
          aria-label="front end developer"
          className="hero__typed-erase"
        >
          Front End Developer
        </p>
      </div>
      <a
        href="#contact"
      >
        <button>
          Let's get in touch
        </button>
      </a>
    </div>
  </section>
  <div
    className="next-button"
  >
    <a
      aria-label="next button"
      href="#about me"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium next-button-icon css-1tzp3k3-MuiSvgIcon-root"
        data-testid="ExpandCircleDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
        />
      </svg>
    </a>
  </div>
  <section
    className="sc-bcXHqe ikGjSH"
    id="about me"
  >
    <div
      className="about"
    >
      <h2
        className="about__title"
      >
        about 
        <strong
          className="about__title-main"
        >
          me
        </strong>
      </h2>
      <p
        aria-label="introduction about oriol raventós"
        className="about__content"
      >
        My name is oriol raventós and I'm a
         
        <strong
          className="about__content-main"
        >
          Junior Front End Developer
        </strong>
         
        based in 
        <strong
          className="about__content-main"
        >
          Barcelona
        </strong>
        . I describe myself as a passionate developer who loves
         
        <strong
          className="about__content-main"
        >
          coding
        </strong>
        , creating new content for the internet and facing new challenges.
      </p>
       
      <br />
      <p
        aria-label="past proffesional experiences"
        className="about__content"
      >
        Not so long ago I used to work as a
         
        <strong
          className="about__content-main"
        >
          SEO Technician
        </strong>
        , but now I work with code.
      </p>
      <br />
       
      <p
        aria-label="free time activities"
        className="about__content"
      >
        Aside from my job, I like to enjoy my free time climbing, running or on tech meetups around Barcelona.
      </p>
    </div>
    <div
      aria-label="oriol raventós profile picture"
      className="image-profile"
    >
      <img
        alt="oriol raventós front end developer"
        src="https://oopugzqermglvumujghs.supabase.co/storage/v1/object/public/images/oriolprofile.webp?t=2022-12-18T17%3A00%3A42.102Z"
      />
    </div>
  </section>
  <section
    className="sc-jSUZER cflCxp"
  >
    <span
      aria-label="loader"
      className="loader"
    />
  </section>
  <section
    className="sc-gKPRtg AeGiM"
    id="projects"
  >
    <h2>
      projects
    </h2>
    <div
      aria-label="projects list"
      className="projects__list"
    />
  </section>
  <section
    className="sc-gswNZR iZgLgD"
    id="contact"
  >
    <h2>
      contact 
      <strong>
        me
      </strong>
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div>
        <input
          aria-label="name"
          autoComplete="off"
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="name"
          type="text"
        />
      </div>
      <div>
        <input
          aria-label="email"
          autoComplete="off"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="e-mail"
          type="email"
        />
      </div>
      <div>
        <input
          aria-label="telephone"
          autoComplete="off"
          id="telephone"
          name="telephone"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="telephone"
          type="number"
        />
      </div>
      <div>
        <textarea
          aria-label="message"
          autoComplete="off"
          id="message"
          name="message"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="message"
        />
      </div>
      <button
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </section>
  <section
    className="sc-dkrFOg hfDpPD"
  >
    <p
      aria-label="created by oriol raventós"
    >
      created by 
      <strong>
        oriol raventós
      </strong>
    </p>
    <div
      className="footer-icons"
    >
      <a
        alt="oriol raventós github profile"
        aria-label="oriol raventós github profile"
        href="https://github.com/oriolraventosc"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </a>
      <a
        alt="oriol raventós linkedin profile"
        aria-label="oriol raventós linkedin profile"
        href="https://www.linkedin.com/in/oriolraventosc/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </a>
      <a
        alt="oriol raventós e-mail"
        aria-label="oriol raventós email"
        href="mailto: oriol.raventos.c@gmail.com"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
      </a>
      <a
        alt="oriol raventós instagram profile"
        aria-label="oriol raventós instagram profile"
        href="https://www.instagram.com/oraventosc/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="InstagramIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
      </a>
      <a
        alt="oriol raventós twitter profile"
        aria-label="oriol raventós twitter profile"
        href="https://twitter.com/oraventosc"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </a>
      <a
        alt="oriol raventós telephone number"
        aria-label="oriol raventós telephone number"
        href="tel: 664622830"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-17z2nsg-MuiSvgIcon-root"
          data-testid="CallIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
          />
        </svg>
      </a>
    </div>
  </section>
</div>
`;
